src/esp/gfx/MaterialUtil.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "MaterialUtil.h"
+
+#include <Magnum/Trade/PbrMetallicRoughnessMaterialData.h>
+#include <Magnum/Trade/PhongMaterialData.h>
+
+namespace Mn = Magnum;
+
+namespace esp {
+namespace gfx {
+
+gfx::PhongMaterialData::uptr buildPhongFromPbrMetallicRoughness(
+    const Mn::Trade::PbrMetallicRoughnessMaterialData& material,
+    int textureBaseIndex,
+    const std::vector<std::shared_ptr<Magnum::GL::Texture2D>>& textures) {
+  // NOLINTNEXTLINE(google-build-using-namespace)
+  using namespace Mn::Math::Literals;
+
+  auto finalMaterial = gfx::PhongMaterialData::create_unique();
+
+  // If there's a roughness texture, we have no way to use it here. The safest
+  // fallback is to assume roughness == 1, thus producing no spec highlights.
+  const float roughness =
```

Comment:
Note that probably 98% of PBR models on the internet use textures for roughness and metalness, so this conversion function is basically useless for them. This function is only effective if you happen to be authoring PBR materials with *constant* metalness and roughness, which happens to be the case for the Zatun assets.